[ESPN] Include additional URL formats (fixes issue #13244)

parmjitv opened the pull request:
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x ] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [ x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

Updated the regex for the [ESPN] extractor to include additional URL formats for videos featured on the site - issue #13244.

Cheers,

Parmjit V.

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -10,7 +10,7 @@
 
 class ESPNIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:espn\.go|(?:www\.)?espn)\.com/video/clip(?:\?.*?\bid=|/_/id/)(?P<id>\d+)'
+    _VALID_URL = r'https?://(?:(?:(\w+\.)+)?espn\.go|(?:www\.)?espn)\.com/(?:video/clip(?:\?.*?\bid=|/_/id/)|watch/player\?.*?\bid=)(?P<id>\d+)'
```
1. `(\w+\.)` is captured for no purpose.
2. It's probably time to make regex verbose `(?x)`.

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'http://nonredline.sports.espn.go.com/video/clip?id=19744672',
```
`only_matching`.

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'https://cdn.espn.go.com/video/clip/_/id/19771774',
```
`only_matching`.

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'http://www.espn.com/watch/player?id=19141491',
```
`only_matching`.

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -25,15 +25,70 @@ class ESPNIE(InfoExtractor):
+        },
+        'expected_warnings': ['Unable to download f4m manifest'],
+    }, {
+        'url': 'http://www.espn.com/watch/player?bucketId=257&id=19505875',
```
`only_matching`.

parmjitv commented:
I have applied the requested fixes, please review at your convenience. Thanks!

parmjitv commented:
Shall I make any further corrections for this PR?  Thank you!

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -10,7 +10,19 @@
+                        )\.com/
+                        (?:
+                            video/clip(?:\?.*?\bid=|/_/id/)|
+                            watch/player\?.*?\bid=
```
Does not cover http://www.espn.com/watch/player/_/id/19141491.

dstftw commented on youtube_dl/extractor/espn.py:
```diff
@@ -10,7 +10,19 @@
+                        )\.com/
+                        (?:
+                            video/clip(?:\?.*?\bid=|/_/id/)|
+                            watch/player(?:\?.*?\bid=|/_/id/)
```
Code duplication.

parmjitv commented:
Good catch - I have updated the regex and added this URL to the test suite for this extractor.  Please let me know if there are any other issues.  Thanks!
